### ISSUE TITLE
About pre-trained models ?

### ISSUE THREAD

**Issue by zhanyuanyang** [ljjcoder/CSEI#1]
Thanks for the excellent work !

https://github.com/ljjcoder/CSEI/blob/1dec6205c234e229629bdc2e88ccc3350f26f620/train_with_inpaint_read_from_data_fixed.py#L118

I am wondering if the pre-trained model used in the previous line was trained on base classes by image classification?

Can you provide the "best_model.pth.tar" ?

**Comment by ljjcoder** [comment]
@Yzy5020 It is trained on Base set which is processed by erasing-inpaiting. you can download the pre-trained model from here: https://drive.google.com/drive/folders/1TdhjO8fWB9nFj_sN2EmZHKpBNFIsn0ns?usp=sharing

**Comment by zhanyuanyang** [comment]
Thanks for your quick reply!

But now I got a problem like this:
![image](https://user-images.githubusercontent.com/23455720/125197237-65242380-e28f-11eb-97e3-e09f3d159efa.png)

I think the problem may be in this line:
https://github.com/ljjcoder/CSEI/blob/1dec6205c234e229629bdc2e88ccc3350f26f620/torchFewShot/dataset_loader/train_inpainting_loader.py#L146

How to fix it?

**Comment by ljjcoder** [comment]
@Yzy5020 I guess you should replace 'train_5/train_5' with 'train_5’

**Comment by zhanyuanyang** [comment]
Thanks for your advice, the problem has been solved.

I have tried ./train.sh by using the parameters "--nExemplars 1" and "--nExemplars 5" respectively.

But the 1-shot & 5-shot results I got are different from the ones in the paper.

![image](https://user-images.githubusercontent.com/23455720/125247213-45d7d580-e325-11eb-9e6d-cb5d57874a04.png)

![image](https://user-images.githubusercontent.com/23455720/125247132-32c50580-e325-11eb-91f1-9b445a5bbdf9.png)

Is this normal?

**Comment by ljjcoder** [comment]
@Yzy5020 
You can refer to my training log
[log_train.txt](https://github.com/ljjcoder/CSEI/files/6800215/log_train.txt).  Due to randomness，it is slightly lower than paper report.

**Comment by Yzy5020** [comment]
OK, thanks a lot, I will try it a few more times.

But I am still confused about the pre-training process?

Can you provide the code for pre-training?

Also, how to use these 3 files?
![image](https://user-images.githubusercontent.com/23455720/125301211-560ba700-e35d-11eb-95ad-33c0d4c0b847.png)

**Comment by zhanyuanyang** [comment]
OK, thanks a lot, I will try it a few more times.

But I am still confused about the pre-training process?

Can you provide the code for pre-training?

Also, how to use these 3 files?
![image](https://user-images.githubusercontent.com/23455720/125301211-560ba700-e35d-11eb-95ad-33c0d4c0b847.png)

**Comment by ljjcoder** [comment]
@Yzy5020 In this code, I fixed all the random number seeds, and the result of running multiple times should be the same. Your result is a bit strange, you get an accuracy of 66.9% at 85 epochs, but only 61.2% in the final test.
you can set only_CSEI=True in train_with_inpaint_read_from_data_fixed.py at line 57  for pretrain.

**Comment by zhanyuanyang** [comment]
![image](https://user-images.githubusercontent.com/23455720/125721205-83658e1e-9927-4a24-8584-a60de7dee449.png)

The result looks normal now.

Thank you very much.
